lib/client-assets.php

```diff
@@ -177,17 +182,37 @@ function gutenberg_override_style( $handle, $src, $deps = array(), $ver = false,
  * @since 4.5.0
  */
 function gutenberg_register_packages_scripts() {
-	$packages_dependencies = include dirname( __FILE__ ) . '/packages-dependencies.php';
+	foreach ( glob( gutenberg_dir_path() . 'build/*/index.js' ) as $path ) {
```

Comment:
Depending if and how similar code would make its way to core, I'm a bit sensitive to the performance implications of a `glob`.

- https://gschoppe.com/wordpress/benchmarking-gutenberg/#comment-3154
- https://www.phparch.com/2010/04/putting-glob-to-the-test/

Seems like there could be optimizations:

- Avoiding the folder scan and pregenerating the scripts at build time (similar to [`generate-gutenberg-php.php`](https://github.com/WordPress/gutenberg/blob/master/bin/generate-gutenberg-php.php))
- Substituting `glob` with something faster (`opendir`)

Reply:
👍A nice enhancement for a future version or simply another part of the build would be to produce what is needed in PHP without needing to hit the filesystem or parse JSON.

This came up during the core-js slack chat.

CC: @gziolo

Reply:
Related conversation on Slack:

https://wordpress.slack.com/archives/C5UNMSU4R/p1556633084220800